### PR TITLE
supernova: add mutex to tcp send function

### PR DESCRIPTION
**Pull request by sonoro1234**
This is as PR #4410 after clang format.

### CODE REVIEW

**Comment by brianlheim** [comment]
thanks!

File: server/supernova/sc/sc_osc_handler.hpp [hunk 1]
```diff
@@ -244,6 +244,7 @@ class sc_osc_handler : private detail::network_thread, public sc_notify_observer
         sc_osc_handler* osc_handler;
         boost::endian::big_int32_t msg_size_;
         std::vector<char> msg_buffer_;
+        std::mutex tcp_mutex;
```
Comment (brianlheim) [inline server/supernova/sc/sc_osc_handler.hpp]: can you rename this to `socket_mutex_`? indicates the relationship more clearly and the trailing underscore is more consistent with the rest of supernova (i think `osc_handler`'s naming is a typo)

Comment (sonoro1234) [reply server/supernova/sc/sc_osc_handler.hpp]: Are you sure?
See: https://github.com/supercollider/supercollider/blob/develop/server/supernova/sc/sc_osc_handler.hpp#L126

Comment (brianlheim) [reply server/supernova/sc/sc_osc_handler.hpp]: sure, it's not consistent, whatever you think is good, but the variable name would be clearer as `socket_mutex` with or without the underscore

Comment (sonoro1234) [reply server/supernova/sc/sc_osc_handler.hpp]: done

**Comment by sonoro1234** [comment]
appveyor is failing due to missisg asiosdk, but I haven't touch that!!!

**Comment by brianlheim** [comment]
known issue, steinberg was doing site maintenance, should be resolved now. sorry for any confusion!

**Comment by dyfer** [comment]
This also solves https://github.com/supercollider/supercollider/issues/4062.

**Comment by sonoro1234** [comment]
Thanks @dyfer I was almost sure!!

**Comment by sonoro1234** [comment]
@dyfer This is a very small PR could you review?

**Comment by dyfer** [comment]
I have tested this from the user perspective on macOS.
- I can confirm that sclang connects to supernova on tcp (solves #4062).
- I have tested requesting large OSC strings from supernova on tcp (see example in #3269) and it works for strings much larger than UDP strings
- I have run a larger project of mine that uses OSC communication extensively for setting buses and getting values from the server - works without issues.

I have discovered two quirks with TCP unrelated to this PR (one with sclang, one with supernova) for which I'll submit separate Issues.

**Comment by sonoro1234** [comment]
Shouldn't it be merged on 3.10.3rc2?

**Pull request merged**